yellowbrick/classifier/prcurve.py

```diff
@@ -0,0 +1,472 @@
+
+            # Compute PRCurve for all classes
+            for idx, cls in enumerate(self.classes_):
+                self.precision_[cls], self.recall_[cls], _ = sk_precision_recall_curve(Y[:,idx], y_scores[:,idx])
+                self.score_[cls] = average_precision_score(Y[:,idx], y_scores[:,idx])
+
+            # Compute micro average PR curve
+            self.precision_[MICRO], self.recall_[MICRO], _ = sk_precision_recall_curve(
+                Y.ravel(), y_scores.ravel()
+            )
+            self.score_[MICRO] = average_precision_score(Y, y_scores, average=MICRO)
+
+        # Draw the figure
+        self.draw()
+
+        # Return a score between 0 and 1
+        if self.target_type_ == BINARY:
+            return self.score_
+        return self.score_[MICRO]
+
+    def draw(self):
+        """
+        Draws the precision-recall curves computed in score on the axes.
+        """
+        if self.iso_f1_curves:
+            for f1 in np.linspace(0.2, 0.8, num=4):
+                x = np.linspace(0.01, 1)
+                y = f1 * x / (2 * x - f1)
+                self.ax.plot(x[y>=0], y[y>=0], color='#333333', alpha=0.2)
+                self.ax.annotate('$f_1={:0.1f}$'.format(f1), xy=(0.9, y[45]+0.02))
+
+        if self.target_type_ == BINARY:
+            return self._draw_binary()
+        return self._draw_multiclass()
+
+    def _draw_binary(self):
+        """
+        Draw the precision-recall curves in the binary case
+        """
+        self._draw_pr_curve(self.recall_, self.precision_, label="binary PR curve")
+        self._draw_ap_score(self.score_)
+
+
+    def _draw_multiclass(self):
+        """
+        Draw the precision-recall curves in the multiclass case
+        """
+        # TODO: handle colors better with a mapping and user input
+        if self.per_class:
+            for cls in self.classes_:
+                precision = self.precision_[cls]
+                recall = self.recall_[cls]
+
+                label = "PR for class {} (area={:0.2f})".format(cls, self.score_[cls])
+                self._draw_pr_curve(recall, precision, label=label)
+
+        if self.micro:
+            precision = self.precision_[MICRO]
+            recall = self.recall_[MICRO]
+            self._draw_pr_curve(recall, precision)
+
+        self._draw_ap_score(self.score_[MICRO])
+
+    def _draw_pr_curve(self, recall, precision, label=None):
+        """
+        Helper function to draw a precision-recall curve with specified settings
+        """
+        self.ax.step(
+            recall, precision, alpha=self.line_opacity, where='post', label=label
+        )
+        if self.fill_area:
+            self.ax.fill_between(recall, precision, step='post', alpha=self.fill_opacity)
+
+    def _draw_ap_score(self, score, label=None):
+        """
+        Helper function to draw the AP score annotation
+        """
+        label = label or "AP={:0.2f}".format(score)
```

Comment:
Mind if we change "AP" to "Avg precision" here for readability?

Reply:
Yeh, I think that looks better, good suggestion!

![binary_precision_recall](https://user-images.githubusercontent.com/745966/45588021-37021780-b8dc-11e8-8c4d-b2db0fcb3d5c.png)

![multiclass_precision_recall](https://user-images.githubusercontent.com/745966/45588022-38334480-b8dc-11e8-8d6d-eeeba402b11d.png)

![multiclass_precision_recall_full](https://user-images.githubusercontent.com/745966/45588026-3b2e3500-b8dc-11e8-98a4-1aa14f38b157.png)

Reply:
except it took me 3 whole minutes to figure out if I had spelled "Precision" correctly  ... it still looks weird to me